### ISSUE TITLE
Any suggestion for speeding up the prediction part?

### ISSUE THREAD

**Issue by tabsun** [MaybeShewill-CV/CRNN_Tensorflow#377]
Thanks to your work and I have already got a good crnn model. In my test the accuracy is satisfying.
But the inference time is a pity: almost 140ms per image with size = 32 x 150. I noticed that some other repos run at < 10ms for the same image input under pytorch framework. Is this normal that tf runs much slower than pytorch? Or something I have make wrong?
Note: I have changed the ctc_beam_search_decoder to ctc_greedy_decoder to get a higher speed.
[The pytorch version crnn](https://github.com/chineseocr/chineseocr)

**Comment by MaybeShewill-CV** [comment]
@tabsun If beam search was used then you may reduce the beam_width params:)

**Comment by tabsun** [comment]
I have replaced ctc_beam_search_decoder with greedy_decoder. As the tensorflow doc says: greedy decoder is a special case when beam_width=1 in beam search decoder but it runs even faster than beam search decoder. 
I wonder if my speed @ 140ms is correct? Or how fast is it in your test?

**Comment by MaybeShewill-CV** [comment]
@tabsun I will test it once I got spare time:)

**Comment by BigPandaCPU** [comment]
> I have replaced ctc_beam_search_decoder with greedy_decoder. As the tensorflow doc says: greedy decoder is a special case when beam_width=1 in beam search decoder but it runs even faster than beam search decoder.
> I wonder if my speed @ 140ms is correct? Or how fast is it in your test?
你好，这里的beam_width我理解是对每一个时刻的输出（该输出是一个num_classes*1的向量，其中的每一个数字表示当前时刻输出属于每一个类别的概率），beam_width=1就表示对当前时刻输出，找到概率最大的那个所在的类别作为当前时刻的输出。当beam_width=10,就表示取前10个作为当前时候输出的候选值。对所有时刻都是进行这样处理，然后在从这些所有时刻的输出中，找到组合输出的label最大的概率，作为整个序列的输出。
当beam_width=1，就是特殊情况，就只有一种组合，时间当然是最快的。而当beam_width=10的时候，就有10^(T)种组合，肯定会耗时的。
另外，可以看考这篇博客，对ctc的理解。https://www.cnblogs.com/skyfsm/p/10335717.html#4508016
